aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -1059,6 +1063,17 @@ module aptos_framework::coin {
             amount
         );
         let withdrawn_coin = if (coin_amount_to_withdraw > 0) {
+            let metadata = paired_metadata<CoinType>();
```

Comment:
I think we can just call ensure_paired_fa_exists, and always use FA permission, instead of requiring master signer otherwise , correct @lightmark  ?